ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,107 @@
+            {
+                if (targetSplit.Contains(word))
+                    result++;
+            }
+
+            return (1.0 - ((double)result / (double)Math.Max(source.Length, target.Length)));
+        }
+        /// <summary>
+        /// Comparing friend lists
+        /// </summary>
+        /// <param name="x">First list</param>
+        /// <param name="y">Second list</param>
+        /// <returns>True if lists are equal, otherwise false</returns>
+        static bool CompareFriends(Friend[] x, Friend[] y)
+        {
+            if(x.Length != y.Length) return false;
+            for (int i = 0; i < x.Length; i++)
+            {
+                if (x[i].Name != y[i].Name)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        static void Main(string[] args)
+        {
+            var people = JsonConvert.DeserializeObject<IEnumerable<Person>>(File.ReadAllText("data.json"));
+
+            var personEast = people.MaxBy(x => x.Longitude);
+            var personNorth = people.MaxBy(x => x.Latitude);
+            var personWest = people.MinBy(x => x.Longitude);
+            var personSouth = people.MinBy(x => x.Latitude);
+
+            Console.WriteLine($"{personNorth.Name} is located farthest north Position: ({personNorth.Longitude}), ({personNorth.Latitude})");
+            Console.WriteLine($"{personSouth.Name} is located farthest south Position: ({personSouth.Longitude}), ({personSouth.Latitude})");
+            Console.WriteLine($"{personWest.Name} is located farthest west Position: ({personWest.Longitude}), ({personWest.Latitude})");
+            Console.WriteLine($"{personEast.Name} is located farthest east Position: ({personEast.Longitude}), ({personEast.Latitude})");
+
+            var distanceMax = people.Select(x => people
+                                    .Select(y => Distance(xLat: x.Latitude, yLat: y.Latitude, xLong: x.Longitude, yLong: y.Longitude))
+                                    .Max()).Max();
+            var distanceMin = people.Select(x => people
+                                    .Select(y => Distance(xLat: x.Latitude, yLat: y.Latitude, xLong: x.Longitude, yLong: y.Longitude)).OrderBy(x => x).Skip(1)
+                                    .Min()).Min();
+
+            Console.WriteLine($"Maximum distance: {distanceMax}");
+            Console.WriteLine($"Minimum distance: {distanceMin}");
+
+            var aboutsMax = people.Select(x => people
+                                  .Select(y => new { similarity = CalculateSimilarity(x.About, y.About), people = (first: x, second: y) })
+                                  .OrderByDescending(x => x.similarity).Skip(1).First())
+                                  .MaxBy(x => x.similarity);
+
+            Console.WriteLine($"{aboutsMax.similarity} {aboutsMax.people.first.Name} {aboutsMax.people.second.Name}");
+
+            var sameFriend = people.SelectMany(x => people
```

Comment:
You should skip calculation with same persons

Reply:
added where before calculating